start.R

```diff
@@ -213,6 +213,14 @@ if (any(c("--testOneRegi", "--debug", "--quick") %in% flags) & "--restart" %in%
   if (gms::getLine() %in% c("Y", "y")) flags <- c(flags, "--reprepare")
 }
 
+# Check if dependencies for a model run are fulfilled
+if (packageVersion("lucode2") >= "0.31.0") {
+  lucode2::checkDeps(action = "stop")
+} else {
+  stop("REMIND requires lucode2 >= 0.31.0, please run the following to update it:\n",
```

Comment:
I like it, but I'd change the message to "lucode2 >= 0.33.0 is required, install it now? y/n"

Reply:
There's actually a caveat: If lucode2 is already loaded in the R session you might get a crash ("lucode2 lazy load database is corrupt" or similar), or still get the old lucode2 functions in the session that the update was installed. There's also a warning message thrown by renv which suggests to restart the R session in such a case. I guess you could check if lucode2 is already loaded, and if not you install the update, otherwise stop with the error.

Reply:
Oh and there's no way to cleanly unload a package in R, so just unloading first won't work. There are functions for unloading, but the documentation even says the only realiable way to unload a package in R is to start a new R session.

Reply:
At this point in the script, lucode2 is always already loaded because it was used before. I don't think the complexity necessary to automatically install the right lucode version is worth the complexity associated (starting a new R session, then stopping the current session) given that people only have to do this *once*.

Reply:
Agree, wasn't aware of this…